include/mqt-core/dd/Operations.hpp

```diff
@@ -22,83 +22,87 @@
 #include <vector>
 
 namespace dd {
-// single-target Operations
-template <class Config>
-qc::MatrixDD
-getStandardOperationDD(const qc::StandardOperation* op, Package<Config>& dd,
-                       const qc::Controls& controls, const qc::Qubit target,
-                       const bool inverse) {
-  GateMatrix gm;
 
-  const auto type = op->getType();
+inline std::tuple<qc::OpType, fp, fp, fp>
```

Comment:
I am not yet fully happy with this fixes kind of return type. In principle, gates could have an arbitrary number of parameters. The fact that currently supported gates onlly use up to three parameters is purely artificial and there are examples of gates with 4+ parameters that we might want to support in the future. Do you see any kind of way of supporting that use case here? Either via variadic lists or vectors of parameters or something completely different..

Maybe it would even be better to take this one step further and represent an operation by its number of qubits, its type, and its number of parameters. Then one could probably transform the code here to a very uniform style and consolidate it quite a bit.

Adding even one step further, this might be worth pulling from the DD submodule to the `ir/operations` module as a whole given the general usefulness of getting the (inverse) matrix for a given StandardOperation.